test/src/test/java/aima/test/unit/search/uninformed/BidirectionalSearchTest.java

```diff
@@ -29,50 +40,57 @@
     @Parameterized.Parameter
     public String searchFunctionName;
 
-    public <A, S> BidirectionalActions<A> searchForActions(Problem<A, S> originalProblem, Problem<A, S> reverseProblem) {
-        BidirectionalSearch<A, S> searchForActionsFunction = new BidirectionalSearch<A, S>();
-        return searchForActionsFunction.apply(originalProblem, reverseProblem);
+    public <A, S> BidirectionalActions<A> searchForActions(Pair<Problem<A, S>, Problem<A, S>> pair) {
```

Comment:
this may seem picky, but what are BidirectionalActions? Actions that work in two directions? Does this name fit the terminology in the book?

> public interface BidirectionalActions<A> {
    /**
     * Path from initial node to meeting node.
     *
     * @return List of actions that from initial state to meeting of two frontiers.

is this an action? what do actions know of meeting nodes? This should be renamed as this is confusing.